pkg/e2e/routemonitors/routemonitors.go

```diff
@@ -237,3 +243,73 @@ func createPlot(title string) *plot.Plot {
 		plot.Label(plot.ErrorLabeler),
 	)
 }
+
+// Extracts the numbers and title from the route monitor plot files
+func (rm *RouteMonitors) ExtractData(baseDir string) error {
+	outputDirectory := filepath.Join(baseDir, "route-monitors")
+	if _, err := os.Stat(outputDirectory); os.IsNotExist(err) {
+		if err := os.Mkdir(outputDirectory, os.FileMode(0755)); err != nil {
+			return fmt.Errorf("error while creating route monitor report directory %s: %v", outputDirectory, err)
+		}
+	}
+
+	for title, pl := range rm.Plots {
+		_ = pl
+
+		// Open the plot file
+		htmlfilePath := filepath.Join(outputDirectory, fmt.Sprintf("%s.html", title))
+		file, err := os.Open(htmlfilePath)
+		if err != nil {
+			log.Printf("Unable to read route monitor plot file; %v\n", err)
+		}
+
+		// Regex to match numbers for the data variable inside the plot file
+		start_regex := regexp.MustCompile(`var data`)
+		datanum_regex := regexp.MustCompile(`[0-9]*\.?[0-9]+,[0-9]*\.?[0-9]+`)
+
+		// Boolean to only scan for numbers once the line `var data` has passed
+		readdata := false
+
+		// This list stores the numbers
+		datalist := make([][]float64, 0)
```

Comment:
Since is this a list of `time, value` pairs, why not:

```go
type Datum struct {
    Time, Value float64
}

datalist := make([]Datum, 0)
```

I think it'd be easier for others trying to understand the structure of the data if each sample had named fields (though I'm not attached to these names).

Reply:
Yup. I've made the changes. Hope the new name's alright though.